surface/model_openapiv3.go

```diff
@@ -282,10 +368,12 @@ func (b *OpenAPI3Builder) typeForSchema(schema *openapiv3.Schema) (kind FieldKin
 					a := items.GetSchemaOrReference()
 					if a[0].GetReference().GetXRef() != "" {
 						return FieldKind_ARRAY, typeForRef(a[0].GetReference().GetXRef()), format
-					} else if a[0].GetSchema().Type == "string" {
-						return FieldKind_ARRAY, "string", format
-					} else if a[0].GetSchema().Type == "object" {
-						return FieldKind_ARRAY, "object", format
+					} else {
+						// Determine whether the items of the array is a known type
+						types := map[string]bool{"string": true, "integer": true, "number": true, "boolean": true, "array": true, "object": true}
+						if types[a[0].GetSchema().Type] {
```

Comment:
When moving `types` out to a global variable, lets also make this part of the `if` chain as an `else if` please!

```go
if a[0].GetReference().GetXRef() != "" {
  ...
} else if types[a[0].GetSchema().Type] {
  ...
}
```

Reply:
Done